pkg/providers/tinkerbell/reconciler/reconciler.go

```diff
@@ -31,6 +31,20 @@ type IPValidator interface {
 	ValidateControlPlaneIP(ctx context.Context, log logr.Logger, spec *c.Spec) (controller.Result, error)
 }
 
+// Scope object for Tinkerbell reconciler.
+type Scope struct {
+	ClusterSpec      *c.Spec
+	isRollingUpgrade bool
+}
+
+// NewScope creates a new Tinkerbell Reconciler Scope.
+func NewScope(clusterSpec *c.Spec) *Scope {
+	return &Scope{
+		ClusterSpec:      clusterSpec,
+		isRollingUpgrade: false,
+	}
+}
+
 // Reconciler for Tinkerbell.
 type Reconciler struct {
```

Comment:
I could be missing something, but why can't we just add the `isRollingUpgrade` bool field to this Tinkerbell Reconciler struct? 

If we are computing isRollout and the desired CAPI as one of the early phases (which seems to be the plan), then that value should persist across phases since we are using the same Reconciler `r` for every phase

Reply:
Guillermo was able to catch that as an issue in a previous PR I was working on. There were issues with thread safety and introducing state to the controller
https://github.com/aws/eks-anywhere/pull/4994/files#r1110388725

Reply:
Ok good to know! That makes sense. Thanks!